internal/api/core/server.go

```diff
@@ -65,25 +66,68 @@ func NewPersesAPI(serviceManager dependency.ServiceManager, persistenceManager d
 		configendpoint.New(cfg),
 		migrateendpoint.New(serviceManager.GetMigration()),
 		validateendpoint.New(serviceManager.GetSchemas(), serviceManager.GetDashboard()),
-		authendpoint.New(persistenceManager.GetUser(), serviceManager.GetCrypto()),
+		authendpoint.New(persistenceManager.GetUser(), serviceManager.GetJWT()),
 	}
 	return &api{
 		apiV1Endpoints: apiV1Endpoints,
 		apiEndpoints:   apiEndpoints,
+		jwtMiddleware: serviceManager.GetJWT().Middleware(func(c echo.Context) bool {
+			return !*cfg.ActivatePermission
+		}),
 	}
 }
 
 func (a *api) RegisterRoute(e *echo.Echo) {
-	a.registerAPIV1Route(e)
+	// First, let's collect every route.
+	// The expecting result is a tree we will need to loop over.
+	groups := a.collectRoutes()
+	// Now let's create a simple struct that will help us to loop over the route tree.
+	type queue struct {
+		parent *echo.Group
+		group  *shared.Group
+	}
+	var queueList []queue
+	for _, g := range groups {
+		queueList = append(queueList, queue{group: g})
+	}
+	// It is our current element on each iteration.
+	var q queue
+	for len(queueList) > 0 {
+		// Let's grab the first element of the queue and remove it so the size of the queue is decreasing.
+		q, queueList = queueList[0], queueList[1:]
+		// Now we need to initialize the echo group that will be used to finally register in the router the different route.
+		var group *echo.Group
+		if q.parent != nil {
```

Comment:
I think if we invert the if we can remove the else statement, wdyt?

Reply:
`parent` is equal to nil at the first iteration, so unfortunately `group = q.parent.Group(q.group.Path)` without checking first `parent` will make the code panic with a null pointer exception

Reply:
Hmm got it.
So we can keep the variable as is and move the `group = q.parent.Group(q.group.Path)` after the expression.
This should work or I'm missing something?

Reply:
I'm not sure I understand what you mean by moving the assignment with the parent group after the expression :s

Reply:
Nvm, it's wrong what I thought about.